mlflow/tracing/export/mlflow_v3.py

```diff
@@ -176,7 +176,37 @@ def _log_trace(self, trace: Trace, prompts: Sequence[PromptVersion]) -> None:
             if trace:
                 add_size_stats_to_trace_metadata(trace)
                 returned_trace_info = self._client.start_trace(trace.info)
+
+                # Collect attachments from all spans and update their references
+                attachments = []
+                attachment_ref_mapping = {}  # old_ref -> new_ref
+
+                for span in trace.data.spans:
+                    if hasattr(span, "_attachments") and span._attachments:
+                        # Update attachment references with real artifact_uri
+                        from mlflow.tracing.utils.artifact_utils import get_artifact_uri_for_trace
+
+                        artifact_uri = get_artifact_uri_for_trace(returned_trace_info)
+
+                        updated_attachments = {}
+                        for old_ref, attachment in span._attachments.items():
+                            # Generate new reference with real artifact_uri
+                            new_ref = attachment.ref(trace.info.trace_id, artifact_uri)
```

Comment:
Is there an easier way to only update the artifact_uri, since you calculated attachment.ref when setting span inputs/outputs?
Even better could we avoid modifying the span inputs/outputs here? If artifact_uri is the only thing that needs to be updated, could we preserve the rest and `artifact_uri` separately and use a helper function to reconstruct during reading?